### PR TITLE
Add a whisper example using Intel Neural Compressor

### PR DESCRIPTION
**Pull request by yuwenzho**
## Describe your changes

Add a whisper example using Intel® Neural Compressor Quantization.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [x] Update documents if necessary.
- [x] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: examples/whisper/README.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ This folder contains a sample use case of Olive to optimize a [Whisper](https://
 Performs optimization pipeline:
 - CPU, FP32: *PyTorch Model -> Onnx Model -> Transformers Optimized Onnx Model -> Insert Beam Search Op -> Insert Pre/Post Processing Ops*
 - CPU, INT8: *PyTorch Model -> Onnx Model -> Transformers Optimized Onnx Model -> Dynamic Quantized Onnx Model -> Insert Beam Search Op -> Insert Pre/Post Processing Ops*
+- CPU, INT8: *PyTorch Model -> Onnx Model -> Transformers Optimized Onnx Model -> Intel® Neural Compressor Dynamic Quantized Onnx Model -> Insert Beam Search Op -> Insert Pre/Post Processing Ops*
```
Comment (trajepl) [inline examples/whisper/README.md:7]: It is better to tell user to install `olive-ai[inc]` if they run this optimization chain.

IIRC, whisper is in our CI pipeline, you may need to add `neural-compressor` into whisper requirements.txt.

**Review by trajepl — DISMISSED** [review]

**Comment by trajepl** [comment]
image
Looks like the test for inc did not be triggered. You may need to update `test_whisper.py`

**Comment by yuwenzho** [comment]
@trajepl It seems that the example test of whisper inc is hanging on Windows. You mentioned in a previous [PR](https://github.com/microsoft/Olive/pull/157#issuecomment-1517010664) that `neural_compressor` related imports are hanging for some reason on the CI pipeline agent. Is this the same issue here? Do you have any suggestions?

**Comment by trajepl** [comment]
> @trajepl It seems that the example test of whisper inc is hanging on Windows. You mentioned in a previous [PR](https://github.com/microsoft/Olive/pull/157#issuecomment-1517010664) that `neural_compressor` related imports are hanging for some reason on the CI pipeline agent. Is this the same issue here? Do you have any suggestions?

Oh, yes, we can skip the inc_int8 test on windows. Please add follow code snipped under `def test_whisper(device_precision)`:
```python
    import platform

    if platform.system() == "Windows" and device_precision[1].startswith("inc_"):
        pytest.skip("Skip Linux containers on Windows host test case.")
```

File: examples/test/test_whisper.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def setup():
     sys.path.remove(example_dir)
 
 
-@pytest.mark.parametrize("device_precision", [("cpu", "fp32"), ("cpu", "int8")])
+@pytest.mark.parametrize("device_precision", [("cpu", "fp32"), ("cpu", "int8"), ("cpu", "inc_int8")])
 def test_whisper(device_precision):
     from olive.workflows import run as olive_run
```
Comment (trajepl) [inline examples/test/test_whisper.py:34]: To skip test on windows.
```python
    import platform

    if platform.system() == "Windows" and device_precision[1].startswith("inc_int8"):
        pytest.skip("Skip Linux containers on Windows host test case.")
```

**Review by trajepl — APPROVED** [review]

**Pull request merged**